App crashes, when searching for a word that starts with a hyphen

When I open a folder and press "search", then enter any searchword that starts with a hyphen ("-anything" for example) the app crashes immediately after pressing enter.

### Expected behavior
Give me a list of messages containing the given phrase.

### Actual behavior
App crashes

### Steps to reproduce
1. Open inbox
2. press the magnifying glass icon
3. enter "-something" and press Enter

### Environment
K-9 Mail version: 5.600

Android version: 8.0.0 / 7.1.1

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

Found this in logs from my previous tests:

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:353)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)
Caused by: android.database.sqlite.SQLiteException: malformed MATCH expression: [-sddd] (code 1)
#################################################################
Error Code : 1 (SQLITE_ERROR)
Caused By : SQL(query) error or missing database.
	(malformed MATCH expression: [-sddd] (code 1))
#################################################################
	at android.database.sqlite.SQLiteConnection.nativeExecuteForCursorWindow(Native Method)
	at android.database.sqlite.SQLiteConnection.executeForCursorWindow(SQLiteConnection.java:1017)
	at android.database.sqlite.SQLiteSession.executeForCursorWindow(SQLiteSession.java:836)
	at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:62)
	at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:152)
	at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:141)
	at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:219)
	at android.database.AbstractCursor.moveToNext(AbstractCursor.java:268)
	at android.database.CursorWrapper.moveToNext(CursorWrapper.java:202)
	at android.database.CursorWrapper.moveToNext(CursorWrapper.java:202)
	at com.fsck.k9.cache.EmailProviderCacheCursor.<init>(EmailProviderCacheCursor.java:46)
	at com.fsck.k9.provider.EmailProvider.query(EmailProvider.java:253)
	at android.content.ContentProvider.query(ContentProvider.java:1063)
	at android.content.ContentProvider.query(ContentProvider.java:1155)
	at android.content.ContentProvider$Transport.query(ContentProvider.java:244)
	at android.content.ContentResolver.query(ContentResolver.java:760)
	at android.content.ContentResolver.query(ContentResolver.java:710)
	at android.content.CursorLoader.loadInBackground(CursorLoader.java:64)
	at android.content.CursorLoader.loadInBackground(CursorLoader.java:54)
	at android.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:315)
	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:69)
	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:64)
	at android.os.AsyncTask$2.call(AsyncTask.java:333)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
```

--- a/app/ui/legacy/src/main/java/com/fsck/k9/ui/messagelist/MessageListLoader.kt
+++ b/app/ui/legacy/src/main/java/com/fsck/k9/ui/messagelist/MessageListLoader.kt
@@ -1,10 +1,14 @@
 package com.fsck.k9.ui.messagelist
 
+import android.annotation.SuppressLint
 import android.content.ContentResolver
 import android.database.Cursor
+import android.database.MatrixCursor
+import android.database.sqlite.SQLiteException
 import android.net.Uri
 import com.fsck.k9.Account
 import com.fsck.k9.Account.SortType
+import com.fsck.k9.K9
 import com.fsck.k9.Preferences
 import com.fsck.k9.fragment.MLFProjectionInfo
 import com.fsck.k9.fragment.MessageListFragmentComparators.ArrivalComparator
@@ -26,6 +30,7 @@ import com.fsck.k9.search.SqlQueryBuilder
 import com.fsck.k9.search.getAccounts
 import java.util.ArrayList
 import java.util.Comparator
+import timber.log.Timber
 
 class MessageListLoader(
     private val preferences: Preferences,
@@ -62,6 +67,7 @@ class MessageListLoader(
         return MessageListInfo(messageListItems, hasMoreMessages)
     }
 
+    @SuppressLint("Recycle")
     private fun loadMessageListForAccount(account: Account, config: MessageListConfig): Cursor? {
         val accountUuid = account.uuid
         val threadId: String? = getThreadId(config.search)
@@ -110,7 +116,17 @@ class MessageListLoader(
 
         val sortOrder: String = buildSortOrder(config)
 
-        return contentResolver.query(uri, projection, selection, selectionArgs, sortOrder)
+        return try {
+            contentResolver.query(uri, projection, selection, selectionArgs, sortOrder)
+        } catch (e: SQLiteException) {
+            Timber.e(e, "Error querying EmailProvider")
+
+            if (K9.DEVELOPER_MODE && e.message?.contains("malformed MATCH expression") == false) {
+                throw e
+            }
+
+            return MatrixCursor(projection)
+        }
     }
 
     private fun getThreadId(search: LocalSearch): String? {

